src/Shared/HttpValidationProblemDetailsJsonConverter.cs

```diff
@@ -60,10 +61,26 @@ public static void WriteProblemDetails(Utf8JsonWriter writer, HttpValidationProb
         writer.WriteStartObject();
         ProblemDetailsJsonConverter.WriteProblemDetails(writer, value, options);
 
+        static string ConvertName(string name, JsonSerializerOptions options)
+        {
+            if (string.IsNullOrEmpty(name) || options?.DictionaryKeyPolicy == null)
+            {
+                return name;
+            }
+
+            var tokens = name.Split(propertySeparator);
```

Comment:
This looks a bit suspicious. - why do we need to split these? What happens if we don't?

Reply:
I'd be impressed if it still worked. I don't think it's too common for JSON properties to have a `.` in them. Worth checking though.

Reply:
I checked and it did not work, only the first part is converted to camelCase.